scripts/generate_index.py

```diff
@@ -0,0 +1,29 @@
+import os
+from pathlib import Path
+source = Path(__file__).resolve().parent.parent / "source"
+source = source.resolve()
+info = {}
+for p in Path(source).rglob("*"):
+    if p.suffix not in ['.md']:
+        continue
+    relpath = os.path.relpath(p.parent, start=source)
+    if "tests" in relpath:
+        continue
+    if p.name in ['index.md']:
+        continue
+    fpath = relpath + '/' + p.name
+    name = None
+    with p.open() as fid:
+        for line in fid:
+            if line.startswith("# "):
+                name = line.replace('# ', '').strip()
```

Comment:
NIT: Is it guaranteed that all the `.md` files have headings that are hyphen delimited?

Reply:
Do you mean whether they start with `# `?  There is a guard for that.  For hyphens, I'm adding them explicitly below.